lib/screens/signup/cubit/signup_cubit.dart

```diff
@@ -11,13 +11,18 @@ class SignupCubit extends Cubit<SignupState> {
   final SignupRepository _signupRepository;
   SignupCubit(this._signupRepository) : super(SignupInitial());
 
-  Future<void> signup(String name, String email, String password, int mobileNumber) async {
+  Future<void> signup(
+      String firstName, String lastName, String email, String mobileNumber, String password) async {
     try {
       emit(SignupLoading());
-      await _signupRepository.signup(name, email, password, mobileNumber);
+      await _signupRepository.signup(firstName, lastName, email, mobileNumber, password);
       emit(SignupSuccess());
     } on NetworkException {
       emit(SignupError(S.networkException));
+    } on ValidationException catch (e) {
+      emit(SignupError(e.description));
+    } on UnknownException {
+      emit(SignupError(S.unknownException));
     }
```

Comment:
add a `Log.s` here as if the function reaches here, its a severe error.
```suggestion
    } catch(e) {
        Log.s(...)
    }
```

Reply:
@werainkhatri  done! we can merge this PR now